### ISSUE TITLE
small dataset - poor performance

### ISSUE THREAD

**Issue by mglowacki100** [Microsoft/LightGBM#380]
I have small but rather wide dataset (100 columns, 150 rows). 
I use mljar.com as rapid prototyping tool (web-based app, that combines xgboost, lightgbm, rf, knn, nn, ...).
I've got similar results in terms of AUC and log-loss for xgboost, rf, extra-trees and random greedy forest, but way worse for lightgbm.
I understand that:
- my description is not detailed enough to investigate now
- in general, there is no "best classifier"
but maybe this is a known issue? LightGBM is faster than xgboost so maybe there is an obvious trade-off for prediction performance for small datasets? 

If it seems to be bug, I'll investigate further and try to replicate this on different data with more details.

**Comment by guolinke** [comment]
@mglowacki100  try set ```min_data=1 min_hess=0```.

**Comment by guolinke** [comment]
the default value of ```min_data=100```. And you data only have 150 rows. So actually it doesn't learn a real model.

**Comment by mglowacki100** [comment]
Thanks!
Is it possible to add smart-defaults for this parameter (set of values depending on number of rows)? 
Less dumb-proof is warning in documentation about small datasets and those parameters.  
LightGBM, has a lot of knobs and as far I see xgboost doesn't have analogous parameter.

**Comment by Neil-Schneider** [comment]
The analogous parameter in xgboost is `min_child_weight`. The default is `1`.

**Comment by Laurae2** [comment]
@Neil-Schneider this is not right, xgboost's `min_child_weight` is analogous to LightGBM's `min_hess`.

**Comment by Neil-Schneider** [comment]
@Laurae2 Sorry to mislead, you are right. 

Is there a difference between the two when all weights are equal to `1`?

**Comment by Laurae2** [comment]
@Neil-Schneider the weight is taken into account when computing gradient statistics, thus the hessian (2nd derivative / 2nd proxy derivative of the loss function) is impacted. When all weights are equal to `1`, then it is the regular hessian for each observation. If you put all weights to `2`, then the hessian is twice the value for each observation. (etc.)

In short: you might think as a weighted hessian sum to compute whether a split should be allowed or not.

**Comment by akaniklaus** [comment]
@guolinke @Neil-Schneider What about a dataset size of 2K? would defining those parameters as 1 and 0 still necessary? What would be suggested values of these parameters? Is there also a minimum threshold of dataset size for LightGBM? For example, can we say that ideally we should have at least 10K rows? What would be the suggested parameters for 10K?

**Comment by guolinke** [comment]
@akaniklaus it depends. `1` is used for the extreme cases. for the 2k dataset, you can use 10/50/100, depends on your `num_leaves`. 
However, for the small dataset, using small `num_leaves` is recommended. 

These is not a threshold, but you should aware to use simple model (small `num_leaves`) is better for small datasets.

**Comment by akaniklaus** [comment]
@guolinke Thanks. The default value is 20 now so I guess that it is allright as I obtained good accuracy.
